haicrypt/cryspr-openssl-evp.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * SRT - Secure, Reliable, Transport
+ * Copyright (c) 2019 Haivision Systems Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+
+/*****************************************************************************
+written by
+   Haivision Systems Inc.
+
+   2019-06-26 (jdube)
+        OpenSSL Direct AES CRYSPR/4SRT (CRYypto Service PRovider for SRT).
+*****************************************************************************/
+
+#ifndef CRYSPR_OPENSSL_H
+#define CRYSPR_OPENSSL_H
+
+#include <openssl/evp.h>	/* PKCS5_xxx() */
+#include <openssl/aes.h>	/* AES_xxx() */
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(OPENSSL_IS_BORINGSSL))
+# include <openssl/modes.h>	/* CRYPTO_xxx() */
+#endif
+#include <openssl/rand.h>
+#include <openssl/err.h>
+#include <openssl/opensslv.h> /* OPENSSL_VERSION_NUMBER */
+
+/* Define CRYSPR_HAS_AESCTR to 1 if this CRYSPR has AESCTR cipher mode
+   if not set it 0 to use enable CTR cipher mode implementation using ECB cipher mode
+   and provide the aes_ecb_cipher method.
+*/
+#define CRYSPR_HAS_AESCTR 1
+
+/* Define CRYSPR_HAS_AESKWRAP to 1 if this CRYSPR has AES Key Wrap
+   if not set to 0 to enable default/fallback crysprFallback_AES_WrapKey/crysprFallback_AES_UnwrapKey methods
+   and provide the aes_ecb_cipher method  .
+*/
+#if 1//(OPENSSL_VERSION_NUMBER < 0x0090808fL) //0.9.8h
+#define CRYSPR_HAS_AESKWRAP 0
+#else
+#define CRYSPR_HAS_AESKWRAP 1
+#endif
```

Comment:
`#if 1` is meant to stay or was forgotten while debugging?

Reply:
Postponed until EVP used for AES-kwrap. Forced to internal implementation in the mean time. Comment should reflect this as Openssl version in comment does not related to EVP API for aes-wrap.

Reply:
Might, but it's not exactly clear. This that you just wrote could be a better comment.

Reply:
Comment updated